File: app/src/main/java/fr/xgouchet/chronorg/data/queriers/ContentQuerier.java
package fr.xgouchet.chronorg.data.queriers;

import android.content.ContentResolver;
import android.support.annotation.NonNull;

import rx.functions.Action1;

/**
 * @author Xavier Gouchet
 */
public interface  ContentQuerier<T> {

    void queryAll(@NonNull ContentResolver contentResolver, @NonNull Action1<T> action);

    void query(@NonNull ContentResolver contentResolver, @NonNull Action1<T> action, int id);

    boolean save(@NonNull ContentResolver contentResolver, @NonNull T item);

    boolean delete(@NonNull ContentResolver contentResolver, @NonNull T item);
}

File: khronorg/src/main/java/fr/xgouchet/khronorg/data/queriers/ContentQuerier.kt
package fr.xgouchet.khronorg.data.queriers

import android.content.ContentResolver
import fr.xgouchet.khronorg.data.query.Query
import io.reactivex.functions.Consumer

/**
 * @author Xavier Gouchet
 */
interface ContentQuerier<T> {

    fun queryAll(contentResolver: ContentResolver, action: Consumer<T>)

    fun queryWhere(contentResolver: ContentResolver, alter: Query, action: Consumer<T>)

    fun save(contentResolver: ContentResolver, item: T): Boolean

    fun delete(contentResolver: ContentResolver, item: T): Boolean
}
